### DISCUSSION TITLE
Use of the NFS example in the repo never works

### DISCUSSION THREAD

**Discussion by elliottecton** [axboe/fio#1674] [Q&A] [answered]
I'm trying to run some benchmarks using the userland NFS module documented in section [1.13.10. I/O engine](https://fio.readthedocs.io/en/latest/fio_doc.html#job-file-format). Every time I try to run the example `nfs.fio` config file [here](https://github.com/axboe/fio/blob/master/examples/nfs.fio), I get the following stack trace:

```text
fio: engine nfs not loadable
fio: failed to load engine
Bad option <nfs_url=nfs://<redacted_ip>/nfs>
fio: job global dropped
```

I installed `libnfs` and rebooted, but I still got the same error. Has anyone run into this? Am I missing a dependency that is not documented? Should I open a bug?

Linux Client Details:

```text
cat /etc/redhat-release
Red Hat Enterprise Linux release 9.3 (Plow)

uname -a
Linux <hostname_redacted> 5.14.0-284.30.1.el9_2.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Aug 25 09:13:12 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux

rpm -qv nfs-utils libnfs
nfs-utils-2.5.4-20.el9.x86_64
libnfs-5.0.2-2.el9.x86_64

fio --version
fio-3.35-33-g50875
```

Thanks in advance, all!

**Comment by axboe** [comment, marked as answer]
Fio doesn't have nfs support in your build, simply installing libnfs isn't enough. You need to install the development package and then rebuild fio. Unrelated, but you never need to reboot for updating/installing new libraries either.

**Reply by elliottecton** [reply]
Thanks, @axboe. At the risk of sounding very ignorant, which development package are you referring to?

**Reply by elliottecton** [reply]
Nevermind. I figured out that you must install `libnfs-devel`, not just the `libnfs` package. On RHEL 9.3, `sudo dnf install -y libnfs-devel`

**Comment by aagi-girish** [comment]
@elliottecton Just faced the same issue and tried to install all the libnfs packages and fio on kernel 5.14 with rocky linux 9.3 with fio 3.35 but no luck. I have built the packages for libnfs and fio from the github repo one after the other and rebooted the server. It doesn't seem to find the nfs engine. Please share your thoughts.

![image](https://github.com/axboe/fio/assets/80010347/b2069217-d278-495e-98b9-d8701f3cf910)

**Comment by aagi-girish** [comment]
Finally!!!

Got it working. But still getting an error code -14.

![image](https://github.com/axboe/fio/assets/80010347/aca49bb8-f727-4a78-849c-33e1d469c0de)

**Reply by Jianshui** [reply]
Hi @aagi-girish 
have you figure out what error code -14 means? I also met the same issue
thanks

**Reply by axboe** [reply]
-14 is -EFAULT
